SPEC/TYPES.md

```diff
@@ -0,0 +1,16 @@
+TYPES API
+=======
+
+A set of data types are exposed directly from the IPFS instance under `ipfs.types`. That way you're not required to import/require the following.
+
+- [`ipfs.types.Buffer`](https://www.npmjs.com/package/buffer)
+- [`ipfs.types.PeerId`](https://github.com/libp2p/js-peer-id)
+- [`ipfs.types.PeerInfo`](https://github.com/libp2p/js-peer-info)
+- [`ipfs.types.multiaddr`](https://github.com/multiformats/js-multiaddr)
+- [`ipfs.types.multibase`](https://github.com/multiformats/multibase)
+- [`ipfs.types.multihash`](https://github.com/multiformats/js-multihash)
+- [`ipfs.types.CID`](https://github.com/ipld/js-cid)
+- [`ipfs.types.crypto`](https://github.com/libp2p/js-libp2p-crypto)
+- [`ipfs.types.dagPB`](https://github.com/ipld/js-ipld-dag-pb)
+- [`ipfs.types.dagCBOR`](https://github.com/ipld/js-ipld-dag-cbor)
```

Comment:
perhaps scope these under IPLD and add all the ones that are supported today?

Reply:
I agree with your idea. However, I would have to add `ipld/js-cid` too. As a consequence, a breaking change would be created in `js-ipfs` [core/index.js#L62](https://github.com/ipfs/js-ipfs/blob/master/src/core/index.js#L62). What do you think?

Regarding adding the other ones, if we decide for creating the specific scope, I think we may add them as well. So, you are referring to all the not archived JS modules at [ipld](https://github.com/ipld?utf8=%E2%9C%93&q=&type=&language=javascript)?

Reply:
That's a valid point, thanks for thinking of that. Let's keep it as it is then.